code/game/objects/effects/effect_system/effects_chem_smoke.dm

```diff
@@ -47,12 +47,14 @@
 /datum/effect_system/smoke_spread/chem
 	var/obj/chemholder
 	var/list/smoked_atoms = list()
+	var/botany_curse = FALSE // this will be true if smoke is produced by botany fruits
 
-/datum/effect_system/smoke_spread/chem/New()
+/datum/effect_system/smoke_spread/chem/New(botany_curse = FALSE)
 	..()
 	chemholder = new
 	chemholder.create_reagents(1000)
 	chemholder.reagents.set_reacting(FALSE) // Just in case
+	botany_curse = botany_curse
```

Comment:
I think you'll need to rename the proc argument because attempting to set a field to a local var just sets the field to itself: https://discord.com/channels/145533722026967040/727448459862343711/1193324382915936367

Reply:
renamed